ui/packages/consul-ui/tests/integration/adapters/binding-rule-test.js

```diff
@@ -0,0 +1,50 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import getNspaceRunner from 'consul-ui/tests/helpers/get-nspace-runner';
+
+const nspaceRunner = getNspaceRunner('binding-rule');
+module('Integration | Adapter | binding-rule', function(hooks) {
+  setupTest(hooks);
+  const dc = 'dc-1';
+  const id = 'slug';
+  test('requestForQueryRecord returns the correct url/method', function(assert) {
+    const adapter = this.owner.lookup('adapter:binding-rule');
+    const client = this.owner.lookup('service:client/http');
+    const expected = `GET /v1/acl/binding-rule/${id}?dc=${dc}`;
+    const actual = adapter.requestForQueryRecord(client.requestParams.bind(client), {
+      dc: dc,
+      id: id,
+    });
+    assert.equal(`${actual.method} ${actual.url}`, expected);
+  });
+  test("requestForQueryRecord throws if you don't specify an id", function(assert) {
+    const adapter = this.owner.lookup('adapter:binding-rule');
+    const client = this.owner.lookup('service:client/http');
+    assert.throws(function() {
+      adapter.requestForQueryRecord(client.url, {
+        dc: dc,
+      });
+    });
+  });
+  test('requestForQueryRecord returns the correct body', function(assert) {
+    return nspaceRunner(
+      (adapter, serializer, client) => {
+        return adapter.requestForQueryRecord(client.body, {
+          id: id,
+          dc: dc,
+          ns: 'team-1',
+          index: 1,
+        });
+      },
+      {
+        index: 1,
+        ns: 'team-1',
+      },
+      {
+        index: 1,
+      },
+      this,
+      assert
+    );
+  });
+});
```

Comment:
I think we need to add `requestForQuery` tests in here, but yeah you're right, if we don't need the 'request a single item' adapter method you can remove the 'single item' tests from here also 👍

Reply:
Updated tests